tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -396,10 +396,46 @@ struct LdgWrapper {
   std::ostream& out_;
 };
 
+template <typename AFF>
+isl::ast_expr buildAccess(AFF access, const CodegenStatementContext& context) {
+  return context.build().access_from(access);
+}
+
+void emitAccess(isl::ast_expr access, const CodegenStatementContext& context) {
+  context.ss << access.to_C_str();
+}
+
 template <typename AFF>
 void emitAccess(AFF access, const CodegenStatementContext& context) {
+  emitAccess(buildAccess(access, context), context);
+}
+
+// Check that the given expression is an access with constant index expressions
+void checkConstantAccess(isl::ast_expr expr) {
+  auto op = expr.as<isl::ast_expr_op>();
+  auto access = op.as<isl::ast_op_access>();
+  TC_CHECK(access);
+  for (int i = 1; i < access.get_n_arg(); ++i) {
+    auto arg = access.get_arg(i);
+    TC_CHECK(arg.as<isl::ast_expr_int>());
```

Comment:
Let's have a description here saying something like "expected constant subscripts, got <..>".  Not sure if AST nodes can be simply converted to strings, no problem if not.